hushline/settings/__init__.py

```diff
@@ -800,4 +823,25 @@ async def alias(username_id: int) -> Response | str:
             profile_form=profile_form,
         )
 
+    @bp.route("/update-directory-intro", methods=["POST"])
+    @admin_authentication_required
+    def update_directory_intro_text() -> Response:
+        form = UpdateDirectoryIntroTextForm()
+        if form.validate_on_submit():
+            raw_intro_text = form.directory_intro_text.data
+            sanitized_intro_text = sanitize_input(raw_intro_text)
+
+            # Store the string representation in the database
+            OrganizationSetting.upsert(
+                key=OrganizationSetting.DIRECTORY_INTRO,
+                value=str(sanitized_intro_text),  # Convert Markup to string
```

Comment:
We can't use `str(sanitized_tex)` because that will return an _escaped_ version of the string, so every time we do this, we will additoinally escape the text.

Try making the input the following string:

```
script -> <script>alert(1)</script>
hash -> #
amp -> &
```

Submit then refresh. Note that we have saved the escaped text to the DB which is not what we want.

`sanitize_input` should just return a `str` and we should manually wrap it for display like `Markup(sanitize_input(foo))` each time, but _not_ when sanitizing it here for DB insertion

Reply:
got it. updated here: 5607dad